src/assets/CollectiblesController.ts

```diff
@@ -466,19 +470,32 @@ export class CollectiblesController extends BaseController<
    * @param address - Hex address of the collectible contract.
    * @param tokenId - The collectible identifier.
    * @param collectibleMetadata - Collectible optional information (name, image and description).
+   * @param detection - Wether the collectible is manually added or auto-detected for address and chainId.
    * @returns Promise resolving to the current collectible list.
    */
   private async addIndividualCollectible(
     address: string,
     tokenId: string,
     collectibleMetadata: CollectibleMetadata,
+    detection?: DetectionParams,
   ): Promise<Collectible[]> {
     // TODO: Remove unused return
     const releaseLock = await this.mutex.acquire();
     try {
       address = toChecksumHexAddress(address);
-      const { allCollectibles, collectibles } = this.state;
-      const { chainId, selectedAddress } = this.config;
+      const { allCollectibles } = this.state;
+      let chainId, selectedAddress;
+
+      if (detection?.autodetected) {
```

Comment:
Will there ever be a case where `detection.autodetected = false`?

Reply:
We don't know. This leaves the options open for a developer to just flip a boolean instead of passing or not a whole object.

Reply:
Maybe make it optional and `true` by default? `if (detection && (detection?.autodetected ?? true) ) {`

Reply:
I don't think its necessary, I do think it could be confusing since the object is already called `detection` which I would assume is being used in the autodetection context and therefore it would make the attribute `autodetected` redundant.

Reply:
I'll remove it and use the presence as condition for branching